src/datasets/load.py

```diff
@@ -637,51 +740,65 @@ def get_module(self) -> DatasetModule:
         split_modules = {
             split: infer_module_for_data_files(data_files_list) for split, data_files_list in data_files.items()
         }
-        module_name, builder_kwargs = next(iter(split_modules.values()))
-        if any((module_name, builder_kwargs) != split_module for split_module in split_modules.values()):
+        module_name, default_builder_kwargs = next(iter(split_modules.values()))
+        if any((module_name, default_builder_kwargs) != split_module for split_module in split_modules.values()):
             raise ValueError(f"Couldn't infer the same data file format for all splits. Got {split_modules}")
         if not module_name:
             raise FileNotFoundError(f"No (supported) data files or dataset script found in {self.path}")
         # Collect metadata files if the module supports them
-        if self.data_files is None and module_name in _MODULE_SUPPORTS_METADATA and patterns != DEFAULT_PATTERNS_ALL:
-            try:
-                metadata_patterns = get_metadata_patterns_locally(base_path)
-            except FileNotFoundError:
-                metadata_patterns = None
-            if metadata_patterns is not None:
-                metadata_files = DataFilesList.from_local_or_remote(metadata_patterns, base_path=base_path)
-                for key in data_files:
-                    data_files[key] = DataFilesList(
-                        data_files[key] + metadata_files,
-                        data_files[key].origin_metadata + metadata_files.origin_metadata,
-                    )
+        supports_metadata = module_name in _MODULE_SUPPORTS_METADATA
+        if self.data_files is None and supports_metadata and patterns != DEFAULT_PATTERNS_ALL:
+            extend_data_files_with_metadata_files_locally(base_path=base_path, data_files=data_files)
+
         module_path, hash = _PACKAGED_DATASETS_MODULES[module_name]
+        if metadata_configs:
+            builder_cls = import_main_class(module_path)
+            builder_config_cls = builder_cls.BUILDER_CONFIG_CLASS
+            default_config_name = metadata_configs.get_default_config_name()
+            builder_configs = []
+            for config_name in metadata_configs:
+                config_data_files = metadata_configs.resolve_data_files_locally(
+                    config_name=config_name,
+                    base_path=self.path,
+                    with_metadata_files=supports_metadata,
+                    allowed_extensions=ALL_ALLOWED_EXTENSIONS,
+                )
+                builder_configs.append(
+                    metadata_configs.get_builder_config(
```

Comment:
Again, I don't think this should be a method `MetadataConfigs` (it should be built here from the `MetadataConfigs`.